app/assets/javascripts/desktop-protocol-handler.js

```diff
@@ -0,0 +1,44 @@
+(function() {
+
+  var timeIndex
+
+  window.addEventListener("blur", function(e)
+  {
+    clearTimeout(timeIndex);
+  });
+
+  $("#js-download-repos-btn").click(function(e)
+  {
+    displayDownloadMessage();
+  });
+
+  $("#js-desktop-open-btn").click(function(e)
+  {
+    e.preventDefault();
+
+    var el = $(this);
+    var iFrame = $('#js-hidden-iframe')[0];
+
+    timeIndex = setTimeout(function() {
+      window.location = "http://www.classroom.github.com/desktop";
+    }, 3000);
+
+    // attempt to open deep link in iframe to avoid exposing link to user
+    iFrame.contentWindow.location.href = el.attr("href");
```

Comment:
Attempting to open the deep link in a hidden iframe so that Safari does not throw an error and display its own page if it does not exist.